mantidimaging/gui/windows/move_stack_dialog/view.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (C) 2022 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+import uuid
+from typing import Dict
+
+from PyQt5.QtWidgets import QLabel, QComboBox
+
+from mantidimaging.gui.mvp_base import BaseDialogView
+from mantidimaging.gui.windows.move_stack_dialog.presenter import MoveStackPresenter, Notification
+
+STRICT_DATASET_ATTRS = ["Sample", "Flat Before", "Flat After", "Dark Before", "Dark After", "Recon"]
+
+
+class MoveStackDialog(BaseDialogView):
+
+    originDatasetName: QLabel
+    originDataType: QLabel
+
+    destinationNameComboBox: QComboBox
+    destinationTypeComboBox: QComboBox
+
+    def __init__(self, parent, origin_dataset_id: uuid.UUID, stack_id: uuid.UUID, origin_dataset_name: str,
+                 origin_data_type: str, is_dataset_strict: Dict[str, bool]):
+        super().__init__(parent, 'gui/ui/move_stack_dialog.ui')
+
+        self.originDatasetName.setText(origin_dataset_name)
+        self.originDataType.setText(origin_data_type)
+        self._origin_dataset_id = origin_dataset_id
+        self.destination_dataset_info = is_dataset_strict
+        self._stack_id = stack_id
+
+        self.parent_view = parent
+        self.presenter = MoveStackPresenter(self)
+
+        self.destinationNameComboBox.addItems([name for name in is_dataset_strict.keys()])
```

Comment:
Would it work to use a `DatasetSelectorWidgetView` here. It might save having to pass in list of datasets. Also it gives the dataset UUID, so might saving having to find the UUID from the name in `MainWindowPresenter._move_stack`